src/Geometry.cpp

```diff
@@ -528,48 +519,49 @@ int Geometry::findFSRId(LocalCoords* coords) {
   std::size_t fsr_key_hash = key_hash_function(getFSRKey(coords));
 
   /* If FSR has not been encountered, update FSR maps and vectors */
-  if (_FSR_keys_map.find(fsr_key_hash) == _FSR_keys_map.end()){
-
-    /* Get the cell that contains coords */
-    Cell* cell = findCellContainingCoords(curr);
-    
-    /* Get the lock */
-    omp_set_lock(_num_FSRs_lock);
-
-    /* Recheck to see if FSR has been added to maps after getting the lock */
-    if (_FSR_keys_map.find(fsr_key_hash) != _FSR_keys_map.end())
-      fsr_id = _FSR_keys_map.at(fsr_key_hash)._fsr_id;
+  if (!_FSR_keys_map.contains(fsr_key_hash)){
+
+    /* Try to get a clean copy of the fsr_id, adding the FSR data 
+       if necessary where -1 indicates the key was already added */
+    fsr_id = _FSR_keys_map.insert_and_get_count(fsr_key_hash, NULL);
+    if( fsr_id == -1)
+    {
+      fsr_data volatile* fsr;
+      do{
```

Comment:
It's interesting to see where all these are.  A lot of them are in bits of code that haven't been touched in years, and Will would probably blame on his early struggles learning C++.  A rather impressive number of them are `if (something != NULL){`, which probably means they were copy and pasted.  And all the CMFD probably means Sam and Will didn't agree on the style guide until after those got written!

In any case, let's pretend that the space before ` {` is standard practice.  And at some point, someone should kick your grep up a notch and go all the way to a  sed or awk command with a replace!